front/lib/swr/swr.ts

```diff
@@ -129,6 +131,20 @@ const resHandler = async (res: Response) => {
 
 export const fetcher = async (...args: Parameters<typeof fetch>) => {
   const [url, config] = args;
+
+  if (USE_PUBLIC_API && url.toString().startsWith("/api")) {
+    const transformedUrl = url.toString().replace("/api", USE_PUBLIC_API);
```

Comment:
The name `USE_PUBLIC_API` is misleading, it looks like it's a boolean whereas it looks like it stores a string value. Also given that this runs on the client-side, if you want such a hack, you should use public env variables and store the base url in it so it's consistent and correct usage.

Reply:
renamed PUBIC_API_OVERRIDE